pkg/objects/file.go

```diff
@@ -278,7 +278,7 @@ func (f File) uploadLargeObject(body io.Reader, hdr schwift.ObjectHeaders, clean
 		DeleteSegments: cleanupOldSegments,
 	})
 	if err == nil {
-		err = lo.Append(body, int64(f.Job.Segmenting.SegmentSize))
+		err = lo.Append(body, int64(f.Job.Segmenting.SegmentSize), hdr.ToOpts())
```

Comment:
Many of the headers in `hdr` only apply to the object as a whole, not to its segments, so we should only give those headers that we really want here (i.e., as of now, only `X-Delete-At`). That would also reduce the amount of surprises when new headers get added to `hdr` in the future.